modules/460-log-shipper/hooks/generate_config.go

```diff
@@ -181,6 +205,99 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
 	},
 }, generateConfig)
 
+type tlsConfigFromSecret struct {
+	CACert     []byte
+	ClientCert []byte
+	ClientKey  []byte
+	KeyPass    []byte
+}
+
+func getTLSConfigFromSecretSnapshot(name string, snapshot []go_hook.FilterResult) (*tlsConfigFromSecret, error) {
+	for _, secret := range snapshot {
+		if secret.(*corev1.Secret).Name == name {
+			s := secret.(*corev1.Secret)
+			return &tlsConfigFromSecret{
+				CACert:     s.Data["ca.pem"],
+				ClientCert: s.Data["crt.pem"],
+				ClientKey:  s.Data["key.pem"],
+				KeyPass:    s.Data["keyPass"],
+			}, nil
+		}
+	}
+	return nil, fmt.Errorf("secret %s not found", name)
+}
+
+func getTLSSecretRef(dest v1alpha1.ClusterLogDestination) string {
+	switch dest.Spec.Type {
+	case "Elasticsearch":
+		if dest.Spec.Elasticsearch.TLS.SecretRef != nil {
+			return dest.Spec.Elasticsearch.TLS.SecretRef.Name
+		}
+	case "Vector":
+		if dest.Spec.Vector.TLS.SecretRef != nil {
+			return dest.Spec.Vector.TLS.SecretRef.Name
+		}
+	case "Loki":
+		if dest.Spec.Loki.TLS.SecretRef != nil {
+			return dest.Spec.Loki.TLS.SecretRef.Name
+		}
+	case "Splunk":
+		if dest.Spec.Splunk.TLS.SecretRef != nil {
+			return dest.Spec.Splunk.TLS.SecretRef.Name
+		}
+	case "Logstash":
+		if dest.Spec.Logstash.TLS.SecretRef != nil {
+			return dest.Spec.Logstash.TLS.SecretRef.Name
+		}
+	case "Socket":
+		if dest.Spec.Socket.TCP.TLS.SecretRef != nil {
+			return dest.Spec.Socket.TCP.TLS.SecretRef.Name
+		}
+	case "Kafka":
+		if dest.Spec.Kafka.TLS.SecretRef != nil {
+			return dest.Spec.Kafka.TLS.SecretRef.Name
+		}
+	}
+	return ""
+}
```

Comment:
We can simplify this to:

```suggestion
func getTLSSecretRef(dest v1alpha1.ClusterLogDestination) string {
	var secretRef *v1alpha1.SecretRef = nil

	switch dest.Spec.Type {
	case "Elasticsearch":
		secretRef = dest.Spec.Elasticsearch.TLS.SecretRef
	case "Vector":
		secretRef = dest.Spec.Vector.TLS.SecretRef
	case "Loki":
		secretRef = dest.Spec.Loki.TLS.SecretRef
	case "Splunk":
		secretRef = dest.Spec.Splunk.TLS.SecretRef
	case "Logstash":
		secretRef = dest.Spec.Logstash.TLS.SecretRef
	case "Socket":
		secretRef = dest.Spec.Socket.TCP.TLS.SecretRef
	case "Kafka":
		secretRef = dest.Spec.Kafka.TLS.SecretRef
	}

	if secretRef != nil {
		return secretRef.Name
	}

	return ""
}
```